File: languages/go/entity/identityV2/bjj_issuer_key.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// BjjIdentityKey represents an identity BJJ key used.
type BjjIdentityKey struct {
	Args IssuerKeyArgs
}

// NewBjjIdentityKey creates a new BjjIdentityKey instance with the provided issuer key arguments.
func NewBjjIdentityKey(args IssuerKeyArgs) BjjIdentityKey {
	return BjjIdentityKey{
		Args: args,
	}
}

func (s BjjIdentityKey) ToProto() *proto.IdentityKey {
	var localKey *proto.LocalKey
	if s.Args.LocalKey != nil {
		localKey = s.Args.LocalKey.ToProto()
	}

	var managedKey *proto.ManagedKey
	if s.Args.ManagedKey != nil {
		managedKey = s.Args.ManagedKey.ToProto()
	}

	return &proto.IdentityKey{
		LocalKey:   localKey,
		ManagedKey: managedKey,
	}
}

File: languages/js/src/entity/identity_v2/bjj_issuer_key.ts
import { IssuerKey as IssuerKeyProto } from "../../bridge/proto/identity_entities_v2";
import { LocalKey, ManagedKey } from "../key";
import { IssuerKey } from "./issuer_key";
import { IssuerKeyArgs } from "./issuer_key_args";

export class BjjIssuerKey implements IssuerKey {
  args: IssuerKeyArgs;

  constructor(key: LocalKey | ManagedKey | string) {
    this.args = new IssuerKeyArgs(key);
  }

  public toProto(): IssuerKeyProto {
    return IssuerKeyProto.fromPartial({
      localKey: this.args.localKey?.toProto(),
      managedKey: this.args.managedKey?.toProto()
    });
  }
}
